packages/react-async/src/provider.tsx

```diff
@@ -15,34 +17,123 @@ interface ConsumerProps<Value> {
 
 export interface AsyncContextType<Value> {
   Context: React.Context<Value | null>;
-  Provider: React.ComponentType<ProviderProps>;
-  Consumer: React.ComponentType<ConsumerProps<Value>>;
-  Preload: React.ComponentType<{}>;
+}
+
+export interface AsyncContextType<Value>
+  extends AsyncComponentType<Value, never, {}, {}, {}> {
+  Context: Context<Value | null>;
+  Provider: ComponentType<ProviderProps>;
+  Consumer: ComponentType<ConsumerProps<Value>>;
 }
 
 export function createAsyncContext<Value>({
   id,
   load,
 }: Options<Value>): AsyncContextType<Value> {
+  const resolver = createResolver({id, load});
   const Context = React.createContext<Value | null>(null);
 
+  function Root() {
```

Comment:
Are we making this return a component exclusively for nicer errors? I don't hate that idea but we should add a comment mentioning that is why if so. (As far as I can tell this function could just return an object literal)

Reply:
Will add a comment, because yeah, it's just for errors.